programs/clearing_house/src/state/events.rs

```diff
@@ -122,6 +122,8 @@ pub enum OrderActionExplanation {
     MarketOrderFilledToLimitPrice,
     MarketOrderAuctionExpired,
     CanceledForLiquidation,
+    OrderFilledWithAMM,
+    OrderFilledWithMatch,
```

Comment:
wont the maker pubkey field reveal this? (default pubkey if amm)

Reply:
i think if `determine_fulfillment_methods` returns both fulfilment methods, an order can be filled with AMM and not have a default pubkey as maker